tests/MongoDB.Analyzer.Tests.Common/DataModel/Fruit.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using MongoDB.Bson.Serialization.Attributes;
+using System.Collections.Generic;
+using MongoDB.Bson;
+
+namespace MongoDB.Analyzer.Tests.Common.DataModel
+{
+    [BsonDiscriminatorAttribute("Apple")]
+    public class Fruit
+    {
+        public Fruit() { }
+
+        [BsonConstructorAttribute("Name", "Weight", "Color")]
+        public Fruit(string Name, double Weight, string Color, int Quantity, double TotalCost)
+        {
+            this.Name = Name;
+            this.Weight = Weight;
+            this.Color = Color;
+            this.Quantity = Quantity;
+            this.TotalCost = TotalCost;
+        }
+
+        [BsonFactoryMethodAttribute("Name", "Weight", "Color")]
+        public void Factory_Method()
+        {
+
+        }
+
+        [BsonIdAttribute]
+        public string Name { get; set; }
+
+        [BsonDefaultValueAttribute(5.5)]
+        public double Weight { get; set; }
+
+        [BsonRequiredAttribute]
+        public string Color { get; set; }
+
+        [BsonIgnoreAttribute]
+        public int Quantity { get; set; }
+
+        [BsonElementAttribute("Cost")]
+        public double TotalCost { get; set; }
+
+        [BsonIgnoreIfDefaultAttribute(true)]
+        public double Volume { get; set; }
+
+        [BsonIgnoreIfNullAttribute(true)]
+        public object Width { get; set; }
+
+        [BsonDateTimeOptionsAttribute]
+        public DateTime ExpiryDate { get; set; }
+
+        [BsonDictionaryOptionsAttribute(Bson.Serialization.Options.DictionaryRepresentation.Document)]
+        public Dictionary<string, string> DictionaryField { get; set; }
+
+        [BsonTimeSpanOptionsAttribute(Bson.BsonType.String)]
+        public TimeSpan TimeSpanField { get; set; }
+
+        [BsonExtraElementsAttribute]
+        public BsonDocument CatchAll { get; set; }
+    }
+
+    [BsonKnownTypesAttribute(new Type[] {typeof(GreenApple) })]
+    public class Apple : Fruit
+    {
+        public string AppleType { get; set; }
+    }
+
+    [BsonDiscriminatorAttribute("GreenApple")]
```

Comment:
Maybe change the discriminator not to match the class name (I not it's not fullname, but still for clarity)?

Reply:
Done